mesonbuild/dependencies/ui.py

```diff
@@ -45,12 +45,20 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
             self.link_args = ['-framework', 'OpenGL']
             # FIXME: Detect version using self.clib_compiler
             return
-        if self.env.machines[self.for_machine].is_windows():
+        elif self.env.machines[self.for_machine].is_windows():
             self.is_found = True
             # FIXME: Use self.clib_compiler.find_library()
             self.link_args = ['-lopengl32']
             # FIXME: Detect version using self.clib_compiler
             return
+        else:
+            links = self.clib_compiler.find_library('GL', environment, [])
+            has_header = self.clib_compiler.has_header('GL/gl.h', '', environment)
+            if links and has_header:
```

Comment:
has_header is actually a bit non-obvious here, but if the header doesn't exist it will return `(False, False)` and the "if" will consider that true.

The issue is that the backend internals return a two-tuple where:
- the first item is bool "did the check succeed"
- the second item is bool "was this a cached lookup"

So what you want here is:

```suggestion
            has_header = self.clib_compiler.has_header('GL/gl.h', '', environment)[0]
            if links and has_header:
```

Reply:
Thanks! I totally missed that.